wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -981,26 +1003,70 @@ public PushResponse push(PushRequest request) throws GitException, UnauthorizedE
 
             @SuppressWarnings("unchecked")
             Iterable<PushResult> list = (Iterable<PushResult>)executeRemoteCommand(remoteUri, pushCommand);
-            for (PushResult pushResult : list) {
-                Collection<RemoteRefUpdate> refUpdates = pushResult.getRemoteUpdates();
-                for (RemoteRefUpdate remoteRefUpdate : refUpdates) {
-                    if (!RemoteRefUpdate.Status.OK.equals(remoteRefUpdate.getStatus())) {
-                        if (RemoteRefUpdate.Status.UP_TO_DATE.equals(remoteRefUpdate.getStatus())) {
-                            return newDto(PushResponse.class).withCommandOutput("Everything up-to-date");
+            PushResult pushResult = list.iterator().next();
+            String currentBranch = getCurrentBranch();
+            return generatePushResponse(request, pushResult, currentBranch);
+        } catch (GitAPIException exception) {
+            return newDto(PushResponse.class).withCommandOutput(exception.getMessage());
+        }
+    }
+
+    private PushResponse generatePushResponse(final PushRequest request, final PushResult result, final String currentBranch) {
+        if (result != null) {
```

Comment:
How about 
`if (result == null) {
    throw new GitException("No push results received")
}
...` ?
You will get rid of one nesting level in this case and exception will be thrown instead of empty response